spaceship/lib/spaceship/client.rb

```diff
@@ -450,6 +450,96 @@ def has_valid_session
       return false
     end
 
+    def do_sirp(user, password, modified_cookie)
+      require 'fastlane-sirp'
+      require 'base64'
+
+      prime_length = 2048
+
+      client = SIRP::Client.new(prime_length)
+      a = client.start_authentication
+
+      data = {
+        a: Base64.strict_encode64(to_byte(a)),
+        accountName: user,
+        protocols: ['s2k', 's2k_fo']
+      }
+
+      response = request(:post) do |req|
+        req.url("https://idmsa.apple.com/appleauth/auth/signin/init")
+        req.body = data.to_json
+        req.headers['Content-Type'] = 'application/json'
+        req.headers['X-Requested-With'] = 'XMLHttpRequest'
+        req.headers['X-Apple-Widget-Key'] = self.itc_service_key
+        req.headers['Accept'] = 'application/json, text/javascript'
+        req.headers["Cookie"] = modified_cookie if modified_cookie
+      end
+
+      puts("Received SIRP signin init response") if Spaceship::Globals.verbose?
+
+      body = response.body
+      iterations = body["iteration"]
+      salt = Base64.strict_decode64(body["salt"])
+      b = Base64.strict_decode64(body["b"])
+      c = body["c"]
+
+      key_length = 32
+      encrypted_password = pbkdf2(password, salt, iterations, key_length)
+
+      m1 = client.process_challenge(
+        user,
+        to_hex(encrypted_password),
+        to_hex(salt),
+        to_hex(b),
+        is_password_encrypted: true
+      )
+      m2 = client.H_AMK
+
+      unless m1
```

Comment:
Since process_challenge returns `false` or a string, it would be more clear if this check was 
```
if m1 == false
```
I assumed this was a nil check at first.